src/Lexer.js

```diff
@@ -204,6 +204,10 @@ module.exports = class Lexer {
         l = token.items.length;
         for (i = 0; i < l; i++) {
           token.items[i].tokens = this.blockTokens(token.items[i].text, [], false);
+          if (token.items[i].tokens.some(t => t.type === 'space')) {
+            token.loose = true;
+            token.items[i].loose = true;
+          }
```

Comment:
This should be part of the list tokenizer. If someone wants to extend the list tokenizer to prevent loose lists they should only need to extend the list tokenizer.

Reply:
This relies on the previous line as well because we need to Lex the child list items first. Can I just put the whole section into the Tokenizer? (i.e., is it ok if a Tokenizer handles the lexing of its own children?)

https://github.com/markedjs/marked/blob/c934790c3fb2e45d4cbabf96c0536ce2e84a1efd/src/Lexer.js#L206-L210

Reply:
Unfortunately, determining whether the list is loose is not as simple as checking for blank lines in the list `raw` either, since code blocks for example can contain blank lines that don't count.

Kinda dumb, but what about assigning the lexer as a property of its tokenizer object so it can access parsing functions?
```
  static lex(src, options) {
    const lexer = new Lexer(options);
    lexer.tokenizer.lexer = lexer;
    return lexer.lex(src);
  }
```

Reply:
This is a tough one. I'm tempted to say that it is fine how it is since someone could create an extension that runs before the list tokenizer that consumes the list to create a non-loose list. But this is a slippery slope if we start moving stuff out of the tokenizers.

I'm thinking moving `this.blockTokens` into the tokenizers would be a better solution since it is available in the extensions.

Either way I think this will be a breaking change.

Reply:
To be honest it would be nice if all of the block tokens that could contain other block tokens would call `blockTokens` inside the tokenizer.

Reply:
> what about assigning the lexer as a property of its tokenizer object so it can access parsing functions?

That seems to be the easiest way to do it without passing all of the lexer functions into each tokenizer.

Reply:
So you are saying it would be nice if Tokenizers handle their own children tokens instead of making the Lexer do it, right?

I'd be ok making a separate PR for that then, but what would be the preferred way of exposing lexer.blocktokens() to the tokenizers?

Reply:
or the Tokenizer class only stores `options` in the constructor. Maybe we should change it so the `this` in tokenizer functions is the `lexer` like in extension tokenizer functions?

Reply:
> we should change it so the `this` in tokenizer functions is the `lexer` like in extension tokenizer functions?

In extensions we used `call` to do that, but my impression is `call` is somewhat slower than being able to just call the tokenizer functions directly.

Is there some other way to change `this` in the tokenizer functions?

Reply:
I am referring to `call` but if it is slower than we can do something like `this.tokenizer.lexer = this;`

I feel like it should be consistent with extensions so maybe we should change extensions as well?

Reply:
Would we want that PR to refactor all the tokenizers with child tokens or should those all be separate PRs? Changing any of them is probably breaking so maybe all at once, no?

Reply:
Ya all at once. Maybe we should keep this PR as is then change it in that PR.

Reply:
Sounds good.